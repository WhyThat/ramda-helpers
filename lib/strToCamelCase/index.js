import {
  pipe,
  replace,
  toLower,
  toUpper,
} from 'ramda';

/**
* Returns string in camelCase style
* @name strToCamelCase
* @memberof RamdaHelper#
* @param {string} str
* @returns {string}
* @example
*
* strToCamelCase('') //=> ''
* strToCamelCase('camelCaseString')  //=> 'camelCaseString'
* strToCamelCase('StrWithUpper')  //=> 'strWithUpper'
* strToCamelCase('str-with-dashes')  //=> 'strWithDashes'
* strToCamelCase('Iam@-random/string')  //=> 'iamRandomString'
*/
export default pipe(
  replace(/^[A-Z]{1}/, toLower),
  replace(/\W+(\w{1})/g, (_all, letter) => toUpper(letter)));
