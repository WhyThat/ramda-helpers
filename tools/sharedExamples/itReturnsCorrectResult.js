export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
