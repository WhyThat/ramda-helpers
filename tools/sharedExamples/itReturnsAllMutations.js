import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
