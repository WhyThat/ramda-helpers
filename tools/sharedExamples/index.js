
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
