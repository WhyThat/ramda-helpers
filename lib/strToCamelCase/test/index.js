import { itReturnsAllMutations } from '../../../tools/sharedExamples';
import strToCamelCase from '../';

describe('strToCamelCase', () => {
  const mutations = [{
    name: 'With empty string',
    props: '',
  }, {
    name: 'With camelCase string',
    props: 'camelCaseString',
  }, {
    name: 'With dashes',
    props: 'str-with-dashes',
  }, {
    name: 'With first char un uppercase',
    props: 'StrWithUpperCase',
  }, {
    name: 'With random string',
    props: 'Iam@-random/string',
  }];
  itReturnsAllMutations(strToCamelCase, mutations);
});
