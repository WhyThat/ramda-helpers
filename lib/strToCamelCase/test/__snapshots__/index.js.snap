// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strToCamelCase With camelCase string returns correct result 1`] = `"camelCaseString"`;

exports[`strToCamelCase With dashes returns correct result 1`] = `"strWithDashes"`;

exports[`strToCamelCase With empty string returns correct result 1`] = `""`;

exports[`strToCamelCase With first char un uppercase returns correct result 1`] = `"strWithUpperCase"`;

exports[`strToCamelCase With random string returns correct result 1`] = `"iamRandomString"`;
